apisix/core/dns/client.lua

```diff
@@ -130,7 +130,7 @@ end
 
 
 function _M.new(opts)
-    opts.ipv6 = true
+    opts.ipv6 = opts.ipv6 or true
```

Comment:
Why not `= local_conf.apisix.enable_ipv6`? And we need a test for it.

Reply:
> Why not = local_conf.apisix.enable_ipv6?

IMO, `= local_conf.apisix.enable_ipv6` is a breaking change, Imagine a scenario where a third party or serverless plugin uses this function to parse ipv6, this change will causes an unexpected behavior, I wander if this is a fictitious demand.
Do let me know WDYT. Thanks.

> And we need a test for it.

Sure, I will add it.

Reply:
We can use
```
if opts.ipv6 == nil then
  ...
end
```
so user can specify the ipv6 option to overwrite the default one.

AFAIK, we don't promise the DNS client always handles IPv6 query and the `enable_ipv6` is enabled by default, so it's not a break change. (Even if users don't  consider it is a bug fix)

Reply:
OK, I will add test case for it in next PR